pkg/registry/generic/matcher.go

```diff
@@ -93,9 +113,10 @@ type Matcher interface {
 	// include the object's namespace.
 	MatchesSingle() (key string, matchesSingleObject bool)
 
-	// TODO: when we start indexing objects, add something like the below:
-	//         MatchesIndices() (indexName []string, indexValue []string)
-	//       where indexName/indexValue are the same length.
+	// If the index with a given name is supported by the matcher,
+	// returns the value for that index and true, otherwise returns
+	// "" and false.
+	Index(indexName string) (indexValue string, indexSupported bool)
```

Comment:
I think you should call this IndexQuery or IndexMatcher or something to make it clear that it is matching the contents of indexes, not producing values for insertion into an index.